File: core/feature57.go
package main

import "math"

func feature57(arguments map[string]interface{}) map[string]interface{} {
	parameters := extractParameters(arguments, []string{"K", "G", "L", "A", "E", "T"})
	if parameters == nil {
		return nil
	}
	result := make(map[string]interface{})
	K := getParameter(parameters, "K")
	G := getParameter(parameters, "G")
	L := getParameter(parameters, "L")
	A := getParameter(parameters, "A")
	E := getParameter(parameters, "E")
	T := getParameter(parameters, "T")
	Rad := math.Pi / 180.0
	N := K / 2.0
	R := G / 2.0
	B := math.Atan((R-N)/L) / Rad
	M := R + E*math.Sin(B*Rad)
	W := 1.0 / math.Tan(B*Rad)
	U := 1.0 / math.Sin(B*Rad)
	V := 180.0 * math.Sin(B*Rad)
	result["SR"] = N * U
	result["BR"] = M * U
	result["KS"] = 2.0 * N * U * math.Sin((V/T)*Rad)
	result["GS"] = 2.0 * M * U * math.Sin((V/T)*Rad)
	pl := make([]interface{}, 0)
	for X := 0.0; X <= T/2.0; X += 1.0 {
		F := X * (360.0 / T)
		Q := M*U - N*(W+math.Tan(A*Rad))/(math.Cos(B*Rad)+math.Tan(A*Rad)*math.Sin(B*Rad)*math.Cos(F*Rad))
		pl = append(pl, Q)
	}
	result["PointsL"] = pl
	return result
}

File: core/feature57.js
function feature57() {
  var parameters = extractParameters(["K", "G", "L", "A", "E", "T"]);
  return feature57_core(parameters);
}

function feature57_core(parameters) {
  if (parameters == null) {
    return null;
  }
  var result = {};
  var K = getParameter(parameters, "K")
  var G = getParameter(parameters, "G")
  var L = getParameter(parameters, "L")
  var A = getParameter(parameters, "A")
  var E = getParameter(parameters, "E")
  var T = getParameter(parameters, "T")
  var Rad = Math.PI / 180.0
  var N = K / 2.0
  var R = G / 2.0
  var B = Math.atan((R - N) / L) / Rad
  var M = R + E * Math.sin(B * Rad)
  var W = 1.0 / Math.tan(B * Rad)
  var U = 1.0 / Math.sin(B * Rad)
  var V = 180.0 * Math.sin(B * Rad)
  result["SR"] = N * U
  result["BR"] = M * U
  result["KS"] = 2.0 * N * U * Math.sin((V / T) * Rad)
  result["GS"] = 2.0 * M * U * Math.sin((V / T) * Rad)
  var pl = [];
  for (var X = 0.0; X <= T / 2.0; X += 1.0) {
    var F = X * (360.0 / T)
    var Q = M * U - N * (W + Math.tan(A * Rad)) / (Math.cos(B * Rad) + Math.tan(A * Rad) * Math.sin(B * Rad) * Math.cos(F * Rad))
    pl.push(Q);
  }
  result["PointsL"] = pl
  return result;
}
